Memory Cards: Implement conversion in Qt

RedPanda4552 opened the pull request:
### Description of Changes
Implements the memory card conversion option in Qt.

### Rationale behind Changes
Memory card conversion was not implemented in the initial move from wx to Qt.

### Suggested Testing Steps
Convert some memory cards, check that games still see their data on the converted cards.

F0bes commented on pcsx2-qt/Settings/MemoryCardConvertDialog.h:
```diff
@@ -0,0 +1,52 @@
+
+	MemoryCardType m_type = MemoryCardType::File;
+	MemoryCardFileType m_fileType = MemoryCardFileType::PS2_8MB;
+};
```
Missing EOF newline

JordanTheToaster commented:
Doesn't appear to work tried to convert a 32MB card to a folder and Burnout 3 saw it as unformatted.

JordanTheToaster reviewed (approved):
All good here now LGTM.

lightningterror reviewed (commented):
Commit can be named properly with prefix.

refractionpcsx2 reviewed (changes requested):
Gave it a quick test, and I don't think the conversion from folder to memcard is as good as it could be.

At the moment, if you have too much in your folder, it spams the console (which the user might not even have open) then it says "conversion complete!" and the user might delete the folder thinking everything is peachy, especially if they don't have the console window open.

What I would like to see:
1. When converting to memcard, check if there's enough space for the current game on the card, if not, start a new card, so like `folder_converted_1.mcd` or something.  
2. Maybe make the message to the user a bit more verbose, like "Due to the size of the folder, multiple memcards had to be created" or something.  I dunno, creative freedom :P

RedPanda4552 commented:
@refractionpcsx2 

- Picklist now filters options based on computed space requirements of the card
- Fixed an issue where a card containing more than 512 individual files cannot be opened (by basically changing the default number of files Windows allows a C family program to open from 512 to 1024, equal to that of Unix platforms)
- Added some notes to the UI about it stalling while processing the conversion
- Conversion menu will now refuse to open if folder memcard exceeds 64 MB of data, since no file types can hold more than that atm

JordanTheToaster commented:
Converted a few of my memory cards including a 32MB card and it seems to read and save fine looks good to me.

Mrlinkwii commented:
Converted a few of my memory cards including a 64MB memcard  , games seem happy

lightningterror reviewed (commented):
Commit names should still be properly named, maybe squash some too.

RedPanda4552 commented:
Squashed.

lightningterror reviewed (commented):
`Qt: Add dialog for memory card conversion.`

Will be better for consistency.

kenshen112 commented:
Shoved my ABSOULTE UNIT cards into the QT converter. Flipped to folder and flipped back to image with no seeming issues. Was able to browse in BIOS and see sizes and order of saves remained the same

refractionpcsx2 commented:
LGTM, nice work.

If you could squash things down and sort out your commit names, we can go ahead and merge :)

RedPanda4552 commented:
Squashed and rebased
